internal/commands/pull/inputs.go

```diff
@@ -109,3 +117,62 @@ func (i *inputs) resolveRemoteApp(ui terminal.UI, clients cli.Clients) (realm.Ap
 
 	return app, nil
 }
+
+func (i *inputs) resolveClient(ui terminal.UI, realmClient realm.Client, groupID, appID string) (map[string]*zip.Reader, error) {
+	compatibleTemplates, ok, err := realmClient.CompatibleTemplates(groupID, appID)
+	if err != nil {
+		return nil, err
+	}
+	if !ok {
+		return nil, nil
+	}
+
+	if i.TemplateID != "" {
+		for _, template := range compatibleTemplates {
+			// This is the ID that they selected
+			if template.ID == i.TemplateID {
+				// Fetch the template
+				templateZip, err := realmClient.ClientTemplate(groupID, appID, template.ID)
+				if err != nil {
+					return nil, err
+				}
+				return map[string]*zip.Reader{template.ID: templateZip}, nil
+			}
+		}
+		return nil, fmt.Errorf("template %s is not compatible with this app", i.TemplateID)
+	}
+
+	if len(compatibleTemplates) != 0 {
```

Comment:
I asked a similar question up top, but is this case (an empty template array) different than when the `400` is returned from our server?  if not, then we probably don't need `bool` returned from the client `CompatibleTemplates` method.

also, you should invert this if to return early and then un-indent the following block

Reply:
👍 on the invert 

 I think this is where I was confused when I made the reply to your initial comment. For CompatibleTemplates, I made the `bool` return here to check whether the app was made with a template or not.